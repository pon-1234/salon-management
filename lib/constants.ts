// UI Constants
export const UI_CONSTANTS = {
  HEADER_HEIGHT: 73,
  DIALOG_MAX_HEIGHT: '90vh',
  DIALOG_Z_INDEX: 50,
} as const

// Status Values
export const RESERVATION_STATUS = {
  CONFIRMED: 'confirmed',
  PENDING: 'pending',
  CANCELLED: 'cancelled',
  MODIFIABLE: 'modifiable',
} as const

export const CUSTOMER_TYPES = {
  REGULAR: '通常顧客',
  VIP: 'VIP顧客',
  NG: 'NG顧客',
} as const

export const PAYMENT_METHODS = {
  CASH: '現金',
  CARD: 'カード',
  TRANSFER: '振込',
} as const

// Default Values
export const DEFAULT_VALUES = {
  PHONE_NUMBER: '090-1234-5678',
  CUSTOMER_POINTS: 100,
  DESIGNATION_FEE: 3000,
  MODIFICATION_TIMEOUT_MINUTES: 30,
} as const

// Business Hours
export const BUSINESS_HOURS = {
  OPEN: 10,
  CLOSE: 24,
  LAST_ORDER: 23.5,
} as const

// Areas and Locations
export const AREAS = [
  '池袋',
  '新宿',
  '渋谷',
  '品川',
  '銀座',
  '上野',
  '秋葉原',
  '六本木',
  '恵比寿',
  '中野',
] as const

export const PREFECTURES = ['東京都', '神奈川県', '埼玉県', '千葉県'] as const

// Marketing Channels
export const MARKETING_CHANNELS = ['WEB', '電話', '紹介', 'リピート', 'SNS'] as const

// Staff Work Status
export const WORK_STATUS = {
  WORKING: '出勤',
  NOT_WORKING: '未出勤',
  REST: '休み',
  NOT_SET: '未設定',
} as const

export type ReservationStatus = (typeof RESERVATION_STATUS)[keyof typeof RESERVATION_STATUS]
export type CustomerType = (typeof CUSTOMER_TYPES)[keyof typeof CUSTOMER_TYPES]
export type PaymentMethod = (typeof PAYMENT_METHODS)[keyof typeof PAYMENT_METHODS]
export type WorkStatus = (typeof WORK_STATUS)[keyof typeof WORK_STATUS]
