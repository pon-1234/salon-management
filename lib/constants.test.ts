import { describe, it, expect } from "vitest";
import * as constants from "./constants";

describe("Constants", () => {
  it("should match the snapshot for all exported constants", () => {
    expect(constants).toMatchSnapshot();
  });
}); 