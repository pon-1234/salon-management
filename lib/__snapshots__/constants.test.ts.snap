// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Constants > should match the snapshot for all exported constants 1`] = `
{
  "AREAS": [
    "池袋",
    "新宿",
    "渋谷",
    "品川",
    "銀座",
    "上野",
    "秋葉原",
    "六本木",
    "恵比寿",
    "中野",
  ],
  "BUSINESS_HOURS": {
    "CLOSE": 24,
    "LAST_ORDER": 23.5,
    "OPEN": 10,
  },
  "CUSTOMER_TYPES": {
    "NG": "NG顧客",
    "REGULAR": "通常顧客",
    "VIP": "VIP顧客",
  },
  "DEFAULT_VALUES": {
    "CUSTOMER_POINTS": 100,
    "DESIGNATION_FEE": 3000,
    "MODIFICATION_TIMEOUT_MINUTES": 30,
    "PHONE_NUMBER": "090-1234-5678",
  },
  "MARKETING_CHANNELS": [
    "WEB",
    "電話",
    "紹介",
    "リピート",
    "SNS",
  ],
  "PAYMENT_METHODS": {
    "CARD": "カード",
    "CASH": "現金",
    "TRANSFER": "振込",
  },
  "PREFECTURES": [
    "東京都",
    "神奈川県",
    "埼玉県",
    "千葉県",
  ],
  "RESERVATION_STATUS": {
    "CANCELLED": "cancelled",
    "CONFIRMED": "confirmed",
    "MODIFIABLE": "modifiable",
    "PENDING": "pending",
  },
  "UI_CONSTANTS": {
    "DIALOG_MAX_HEIGHT": "90vh",
    "DIALOG_Z_INDEX": 50,
    "HEADER_HEIGHT": 73,
  },
  "WORK_STATUS": {
    "NOT_SET": "未設定",
    "NOT_WORKING": "未出勤",
    "REST": "休み",
    "WORKING": "出勤",
  },
}
`;
